File: src/types/savedSearch.ts
import type { SearchMode } from "./search";

/**
 * 保存検索（スマートビュー）。Rust 側 models::saved_search::SavedSearch と
 * フィールド名を一致させるため snake_case を用いる。
 */
export interface SavedSearch {
  id: number;
  name: string;
  query: string;
  mode: SearchMode;
  sort_order: number;
  created_at: string;
}

File: src-tauri/src/models/saved_search.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct SavedSearch {
    pub id: i64,
    pub name: String,
    pub query: String,
    pub mode: String,
    pub sort_order: i64,
    pub created_at: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateSavedSearchRequest {
    pub name: String,
    pub query: String,
    pub mode: String,
}
